src/main/java/com/tritoncubed/launchapi/DBServlet.java

```diff
@@ -23,29 +24,41 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
         String utc = request.getParameter("utc");
         if (utc == null) {
             response.setStatus(400);
-            response.getWriter().append("Requires utc field.\n");
+            response.getWriter().append("{\"error\" : \"No UTC value.\" }");
         } else {
             Payload payload = LaunchDB.get(Long.parseLong(utc));
-            response.getWriter().append(String.format("{temp: %s}\n", payload.getTemp()));
+            response.getWriter().append(payload.toString());
         }
     }
 
+    
+    
     protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-        Payload payload = new Payload();
-
-        String utc = request.getParameter("utc");
-        if (utc == null) {
-            utc = Long.toUnsignedString(Calendar.getInstance().getTimeInMillis());
-        }
-
-        String temp = request.getParameter("temp");
-        if (temp == null) {
-            response.setStatus(400);
-            response.getWriter().append("Requires temp field.\n");
-        } else {
-            payload.setUtc(Long.parseLong(utc));
-            payload.setTemp(temp);
-            LaunchDB.put(payload);
-        }
+    	byte[] raw = request.getInputStream().readAllBytes(); // Potential problem for large amounts of data (File channel?)
```

Comment:
Won't be a problem for us, but you can explore alternatives for obtaining the bytes into the buffer